src/message.c

```diff
@@ -460,37 +573,57 @@ int message_get_delivery_annotations(MESSAGE_HANDLE message, annotations* delive
 	return result;
 }
 
-int message_set_message_annotations(MESSAGE_HANDLE message, annotations message_annotations)
+int message_set_message_annotations(MESSAGE_HANDLE message, message_annotations annotations)
 {
 	int result;
 
-	if ((message == NULL) ||
-		(message_annotations == NULL))
+	if (message == NULL)
 	{
-		LogError("Bad arguments: message = %p, message_annotations = %p",
-			message, message_annotations);
+		/* Codes_SRS_MESSAGE_01_044: [ If `message` is NULL, `message_set_message_annotations` shall fail and return a non-zero value. ]*/
+		LogError("NULL message");
 		result = __FAILURE__;
 	}
 	else
 	{
-		annotations new_message_annotations;
-
-		new_message_annotations = annotations_clone(message_annotations);
-		if (new_message_annotations == NULL)
-		{
-			LogError("Cannot clone message annotations");
-			result = __FAILURE__;
-		}
-		else
+		if (annotations == NULL)
 		{
+			/* Codes_SRS_MESSAGE_01_145: [ If `annotations` is NULL, the previously stored message annotations associated with `message` shall be freed. ]*/
 			if (message->message_annotations != NULL)
 			{
 				annotations_destroy(message->message_annotations);
```

Comment:
A very good find btw!!!

---
In reply to: [120474861](https://github.com/Azure/azure-uamqp-c/pull/128#discussion_r120474861) [](ancestors = 120474861,120403505)